picasso/src/main/java/com/squareup/picasso/MarkableInputStream.java

```diff
@@ -33,16 +33,26 @@
   private long reset;
   private long limit;
   private long defaultMark = -1;
+  private boolean dynamicLimitEnabled = false;
```

Comment:
I have no interest in making this class generic. Eliminate this boolean and always perform this behavior.